src/cryptonote_basic/connection_context.h

```diff
@@ -55,6 +55,37 @@ namespace cryptonote
       state_normal
     };
 
+    /*
+      This class was originally from the EPEE module. It is identical in function to std::atomic<uint32_t> except
+      that it has copy-construction and copy-assignment defined, which means that earliers devs didn't have to write
+      custom copy-contructors and copy-assingment operators for the outer class, cryptonote_connection_context.
+      cryptonote_connection_context should probably be refactored because it is both trying to be POD-like while
+      also (very loosely) controlling access to its atomic members.
+    */
+    class copyable_atomic: public std::atomic<uint32_t>
```

Comment:
That's cool that you took it over, but I'd ask moving it to the `common` dir and giving it its own source file. At least a header file, if header file + a .cpp file is too much of a change for now.

Reply:
What I was going to do originally was rewrite the copy construction / copy assignment for `cryptonote_connection_context`, because that's the "correct" way to use atomic members. There's a reason that `std::atomic` has its copy construction and assignment deleted. The issue is that that creates a new function that you have to update when you add or remove members from `cryptonote_connection_context`.

Ideally, cryptonote_connection_context would be refactored to not have any atomic members, and instead be accessed through a mutex (for example).  That would take a decent amount of refactoring of pretty critical code that I am not as familiar with. My idea was that this specific change would not permanent if it got merged, and would get removed completely somewhere down the line

Reply:
I understand.